bottlecap/src/secrets/decrypt.rs

```diff
@@ -24,11 +24,39 @@ pub async fn resolve_secrets(config: Arc<Config>, aws_config: &AwsConfig) -> Opt
                     return None;
                 }
             };
+            let mut cloned_config = aws_config.clone();
```

Comment:
`cloned_config` is too confusing, this method has a `config` and `aws_config`, yet this `cloned_config` is a clone of `aws_config`

Do we even need the `cloned_config`?

Can't we then just pass a clone below to `decrypt_aws_*` instead of cloning and then using it in this if?

Reply:
I debated making the aws_config mutable, maybe that's a better option instead of cloning it. I believe we fully consume the aws_config anyway in this case, it's not used by other functions.